oracle/types/market_test.go

```diff
@@ -5,229 +5,216 @@ import (
 		{
-			name:         "empty off-chain ticker",
-			providerName: "test",
-			configs: types.TickerToProviderConfig{
-				constants.BITCOIN_USD: {
-					Name:           "test",
-					OffChainTicker: "",
+			name:     "single market for the provider",
+			provider: "test",
+			market: mmtypes.MarketMap{
+				Markets: map[string]mmtypes.Market{
+					"BTC/USD": {
+						Ticker: mmtypes.NewTicker("BTC", "USD", 8, 1),
+						ProviderConfigs: []mmtypes.ProviderConfig{
+							{
+								Name:           "test",
+								OffChainTicker: "BTC/USDT",
+								Metadata_JSON:  "{}",
+							},
+						},
+					},
 				},
 			},
-			expErr: true,
-		},
-		{
-			name:         "invalid ticker",
-			providerName: "test",
-			configs: types.TickerToProviderConfig{
-				mmtypes.NewTicker("BTC", "USD", 8, 0): {
-					Name:           "test",
-					OffChainTicker: "BTC-USD",
-				},
+			expected: []types.ProviderTicker{
+				types.NewProviderTicker(
+					"test",
+					"BTC/USDT",
+					"{}",
+					types.DefaultTickerDecimals,
+				),
 			},
-			expErr: true,
+			err: false,
 		},
 		{
-			name:         "valid configs",
-			providerName: "test",
-			configs: types.TickerToProviderConfig{
-				constants.BITCOIN_USD: {
-					Name:           "test",
-					OffChainTicker: "BTC-USD",
-				},
-				constants.BITCOIN_USDC: {
-					Name:           "test",
-					OffChainTicker: "BTC-USDC",
+			name:     "provider has no configs in market map",
+			provider: "test",
+			market: mmtypes.MarketMap{
+				Markets: map[string]mmtypes.Market{
+					"BTC/USD": {
+						Ticker: mmtypes.NewTicker("BTC", "USD", 8, 1),
+						ProviderConfigs: []mmtypes.ProviderConfig{
+							{
+								Name:           "other",
+								OffChainTicker: "BTC/USDT",
+								Metadata_JSON:  "{}",
+							},
+						},
+					},
 				},
 			},
-			expErr: false,
+			expected: nil,
+			err:      false,
 		},
 		{
-			name:         "mismatch provider name and config",
-			providerName: "test",
-			configs: types.TickerToProviderConfig{
-				constants.BITCOIN_USD: {
-					Name:           "invalid",
-					OffChainTicker: "BTC-USD",
+			name:     "duplicate markets for the provider",
+			provider: "test",
+			market: mmtypes.MarketMap{
+				Markets: map[string]mmtypes.Market{
+					"ETH/USD": {
+						Ticker: mmtypes.NewTicker("ETH", "USD", 8, 1),
+						ProviderConfigs: []mmtypes.ProviderConfig{
+							{
+								Name:           "test",
+								OffChainTicker: "ETH/USDT",
+								NormalizeByPair: &pkgtypes.CurrencyPair{
+									Base:  "USDT",
+									Quote: "USD",
+								},
+								Metadata_JSON: "{}",
+							},
+						},
+					},
+					"USDT/USD": {
+						Ticker: mmtypes.NewTicker("USDT", "USD", 8, 1),
+						ProviderConfigs: []mmtypes.ProviderConfig{
+							{
+								Name:           "test",
+								OffChainTicker: "ETH/USDT",
+								Invert:         true,
+								NormalizeByPair: &pkgtypes.CurrencyPair{
+									Base:  "ETH",
+									Quote: "USD",
+								},
+								Metadata_JSON: "{}",
```

Comment:
If this field was different from the metadata json for the other ProviderConfig, what is the expected behavior?